src/scripts/clipperUI/components/previewViewer/pdfPreview.tsx

```diff
@@ -22,18 +22,35 @@ import { PreviewViewerPdfHeader } from "./previewViewerPdfHeader";
 import * as _ from "lodash";
 
 interface PdfPreviewState {
-	fadeIn: boolean;
+	showPageNumbers: boolean;
 }
 
 class PdfPreview extends PreviewComponentBase<PdfPreviewState, ClipperStateProp> {
-	private static scrollListenerAdded: boolean = false; // done on purpose
+	private static scrollListenerAdded: boolean = false;
+	private static scrollListenerTimeout: number;
 
 	private addScrollListener() {
 		if (!PdfPreview.scrollListenerAdded) {
 			let previewContentContainer = document.getElementById("previewContentContainer");
 			if (!!previewContentContainer) {
+				// When we detect a scroll, show page numbers immediately.
+				// When the user doesn't scroll for some period of time, fade them out.
 				previewContentContainer.addEventListener("scroll", (ev) => {
-					console.log("scroll handler");
+					if (Utils.isNumeric(PdfPreview.scrollListenerTimeout)) {
```

Comment:
Hmm I looked through what debounce does, and I don't think it's suitable here, at least if we try using it, it will look less readable (and possibly less performant).